---
title: "Intro_R_2"
author: "Tom McBrien"
date: "October 14, 2015"
output: html_document
---


```{r locating data}
d <- read.csv("Dataset_S1.txt", sep = ',', header = TRUE) #reading in data, separate with comma because that's how it's delimited
head(d) #checking data
colnames(d)[12] <- "percent.GC" #renaming column 12 because it was uploaded with a "%" sign as column header, which R just changes to "X"
d$diversity <- d$Pi / (10*1000) #rescale the vector "Pi" into something more useable
d$cent <- d$start >= 25800000 & d$end <= 29700000 #Created new column called "cent" which identifies whether each row is in the centromere or not based on positions given to us by Vince Buffalo's book.
```

```{r looking at SNPs}
d$total.SNPS >= 85 #Taking a look at which values of the vector conform to our specs
d[d$total.SNPs >= 85, ] #This command will extract rows with greater than or equal to 85 SNPs
snpsGT85 <- d[d$total.SNPs >= 85,] #subset data into new data frame
```

```{r subset based on columns}
d[d$Pi > 16 & d$percent.GC > 80, c("start", "end", "depth", "Pi")] #we subset our data based on certain characteristics in the columns and only certain coluns
```

```{r summarize data after subsetting}
summary(d$depth[d$percent.GC >= 80]) 
summary(d$depth[d$percent.GC < 80])
```

```{r subset using subset function}
subset(d, Pi > 16 & percent.GC > 80, c(start, end, Pi, percent.GC, depth)) #subset "d" based on the Pi iand percent.GC data values with certain columns
```

```{r installing ggplot2}
install.packages("ggplot2") #installing ggplot2
library(ggplot2) #important to actually call it on the machine
```

```{r making a basic plot with ggplot2 and dealing with oversaturation}
d$position <- (d$end + d$start) / 2 #making a midpoint
ggplot(d) + geom_point(aes(x=position, y=diversity)) #calling ggplot on dataframe and then adding a layer in which you plot via the geometric point function based on plotting diversity values by their position
ggplot(d, aes(x=position, y=diversity)) + geom_point() #trying to clear up overplotting, not much luck
ggplot(d) + geom_point(aes(x=position, y=diversity, color=cent)) #trying this again but now the aesthetic value includes a coloring based on the "cent" value we made before that tells us whether something is in the centromere or not
ggplot(d) + geom_density(aes(x=diversity), fill="black") #a density plot with a filled color
ggplot(d) + geom_density(aes(x=diversity, fill=cent), alpha=0.4) #density plot showing both non-centromere and centromere datasets with their corresponding diversities. alpha=0.4 makes the graphs more transparent so you can see both
```

```{r checking normality and increasing resolution of data by binning}
ggplot(d) + geom_bar(aes(x=percent.GC)) #checking out normality of data
d$GC.binned <- cut(d$percent.GC, 5) #using cut to bin the percent GC into 5 bins 
ggplot(d) + geom_bar(aes(x=GC.binned)) #checking out our bins
ggplot(d) + geom_density(aes(x=depth, linetype=GC.binned), alpha=0.5) #plotting data this way shows us that both the lowest and highest GC content windows have lower depth
```

```{r merging and combining data}
mtfs <- read.delim("motif_recombrates.txt", header = TRUE) #read in motif recombination rates data
head(mtfs) #checking work
rpts <- read.delim("motif_repeats.txt", header=TRUE) #read in motif repeats
head(rpts)
mtfs$pos <- paste(mtfs$chr, mtfs$motif_start, sep="-") #make a column of specific positions per chromosome in the mtfs dataframe, ultimate gonna use this to connect mtfs and rpts
head(mtfs)
rpts$pos <- paste(rpts$chr, rpts$motif_start, sep="-") #make a column of specific positions per chromosome in the rpts dataframe
head(rpts)
recm <- merge(mtfs, rpts, by.x="pos", by.y="pos") #merge statement, making a new dataset from these two dataframes
head(recm) #take a look
nrow(recm) #how many rows?
recm <- merge(mtfs, rpts, by.x="pos", by.y="pos", all.x=TRUE) #the merge statement with a left outer join, meaning you merge but including all records from the first file (the x position, or mtfs), not just the ones where they line up.
head(recm)
nrow(recm)
```